### PR TITLE
Include TerminalReporter in reference docs #6649

### LINKED ISSUE #6649: doc: TerminalReporter is not documented
It could be added as follows, but only `rewrite` has a docstring and would be included.

It is used with the `pytest_terminal_summary` hook at least, and therefore is public API it seems.

```diff
diff --git i/doc/en/reference.rst w/doc/en/reference.rst
index 50e32d660..552ad895b 100644
--- i/doc/en/reference.rst
+++ w/doc/en/reference.rst
@@ -838,6 +838,13 @@ Session
     :members:
     :show-inheritance:

+TerminalReporter
+~~~~~~~~~~~~~~~~
+
+.. autoclass:: _pytest.terminal.TerminalReporter
+    :members:
+    :inherited-members:
+
 TestReport
 ~~~~~~~~~~

```

### PR DESCRIPTION
**Pull request by soxofaan**
Closes #6649 

- [x] Include documentation when adding new features.
- [x] Include new tests or update existing tests when applicable.
- [X] Allow maintainers to push and squash when merging my commits. Please uncheck this if you prefer to squash the commits yourself.
- [x] Add text like ``closes #XYZW`` to the PR description and/or commits (where ``XYZW`` is the issue number). See the [github docs](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more information.
- [x] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/main/changelog/README.rst) for details.
- [x] Add yourself to `AUTHORS` in alphabetical order.

### CODE REVIEW

**Comment by soxofaan** [comment]
I'm wondering if I should/can  add an import of TerminalReporter somewhere around
https://github.com/pytest-dev/pytest/blob/6c806b499ddbb844753b5c8c4d70a8b98b9d1c3a/src/pytest/__init__.py#L70-L72

**Review by nicoddemus — CHANGES_REQUESTED** [review]
Hi @soxofaan, thanks for the PR. 

The `TerminalReporter` not being part of the docs is an oversight, given they are part of the signature of `pytest_terminal_summary`.


> I'm wondering if I should/can add an import of TerminalReporter somewhere around

Yes we should also publish it as part of the public API: if it is used by a public hook, it should also be available from the `pytest` namespace.

Also [the build](https://readthedocs.org/projects/pytest/builds/25114166/) failed with:

```
/home/docs/checkouts/readthedocs.org/user_builds/pytest/envs/12661/lib/python3.12/site-packages/_pytest/terminal.py:docstring of _pytest.terminal.TerminalReporter.build_summary_stats_line:11: WARNING: Definition list ends without a blank line; unexpected unindent. 
```

Probably just need to fix the docstring here:

https://github.com/pytest-dev/pytest/blob/6c806b499ddbb844753b5c8c4d70a8b98b9d1c3a/src/_pytest/terminal.py#L1342-L1343

To:

```diff
         This function builds a list of the "parts" that make up for the text in that line, in
-        the example above it would be:
+        the example above it would be::
```

---

It is unfortunate that the entire terminal plugin is being passed along to this hook, but too late for this now so we might as well document it.

**Comment by RonnyPfannschmidt** [comment]
In stoll hoping we can replace it,

It's api is from python 2.4 times and libraries more dedicated to the topic recently showed a much nicer variant (like rich)

**Comment by soxofaan** [comment]
thanks for the feedback. All tests pass now

**Comment by nicoddemus** [comment]
> It's api is from python 2.4 times and libraries more dedicated to the topic recently showed a much nicer variant (like rich)

I know, let's hope we manage to eventually find the time (and will) to do that. :crossed_fingers:

**Review by nicoddemus — APPROVED** [review]

File: changelog/6649.doc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Added `TerminalReporter` to `pytest` root namespace and to API reference documentation page.
```
Comment (nicoddemus) [inline changelog/6649.doc.rst:1]:
```suggestion
Added :class:`pytest.TerminalReporter` to the :ref:`api-reference` documentation page.
```

**Comment by nicoddemus** [comment]
This while small should not be backported, as it is introducing a new public symbol, so it should happen in the next minor release.

**Comment by nicoddemus** [comment]
@RonnyPfannschmidt `build_summary_stats_line` now appears in the docs... want to mark it as private instead (using :private: in the docstring)?

**Comment by nicoddemus** [comment]
(I have a local fix for the docs, will push shortly)

**Pull request merged**